src/main/java/mezz/jei/gui/ghost/GhostIngredientDrag.java

```diff
@@ -75,22 +78,35 @@ public void drawItem(Minecraft minecraft, PoseStack poseStack, int mouseX, int m
 				float percentOfDim = Math.min(1, distanceSq / (float) screenDim);
 				lineWidth = 1 + ((1 - (percentOfDim)) * 3);
 			}
-			GL11.glDisable(GL11.GL_TEXTURE_2D);
+
+			RenderSystem.disableTexture();
 			RenderSystem.disableDepthTest();
+			RenderSystem.depthMask(false);
+
+			var oldShader = RenderSystem.getShader();
+			RenderSystem.setShader(GameRenderer::getPositionColorShader);
+
 			GL11.glLineWidth(lineWidth);
 			GL11.glEnable(GL11.GL_LINE_SMOOTH);
 			GL11.glHint(GL11.GL_LINE_SMOOTH_HINT, GL11.GL_NICEST);
-			GL11.glBegin(GL11.GL_LINES);
+
+			var tesselator = RenderSystem.renderThreadTesselator();
+			var builder = tesselator.getBuilder();
+			builder.begin(VertexFormat.Mode.DEBUG_LINES, DefaultVertexFormat.POSITION_COLOR);
```

Comment:
Is there a meaningful difference between `Mode.DEBUG_LINES` and `Mode.LINES`. At a quick glance it looks like there might be, but I am not aware of a normal `DEBUG_LINES` GL mode so this seems to be a mojang specific thing.

Reply:
Yes. DEBUG_LINES maps to GL LINES, while Mode.LINES maps to Mojangs own line tessellation.